great_expectations/experimental/metric_repository/metrics.py

```diff
@@ -56,17 +68,98 @@ def update_forward_refs(cls):
             Batch=Batch,
         )
 
+    @property
+    def value_type(self) -> str:
+        type_ = self.__orig_class__.__args__[0]
+        string_rep = str(type_)
+        if string_rep.startswith("<class"):
+            return type_.__name__
+        else:
+            return string_rep
+
+    @property
+    def metric_type(self) -> str:
+        return self.__class__.__name__
+
+    @classmethod
+    def _get_properties(cls):
+        """in pydandic v2 we can use computed_field.
+        https://docs.pydantic.dev/latest/usage/computed_fields/"""
+        properties = [
+            prop for prop in cls.__dict__ if isinstance(cls.__dict__[prop], property)
+        ]
+        return properties
+
+    def dict(  # noqa: PLR0913
+        self,
+        *,
+        include: AbstractSetIntStr | MappingIntStrAny | None = None,
+        exclude: AbstractSetIntStr | MappingIntStrAny | None = None,
+        by_alias: bool = False,
+        skip_defaults: Optional[bool] = None,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+    ) -> Dict[str, Any]:
+        """Override the dict function to include @property fields, in pydandic v2 we can use computed_field.
+        https://docs.pydantic.dev/latest/usage/computed_fields/
+        """
+        attribs = super().dict(
+            include=include,
+            exclude=exclude,
+            by_alias=by_alias,
+            skip_defaults=skip_defaults,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+        )
+        props = self._get_properties()
+
+        # Include and exclude properties
+        if include:
+            props = [prop for prop in props if prop in include]
+        if exclude:
+            props = [prop for prop in props if prop not in exclude]
+
+        # Update the attribute dict with the properties
+        if props:
+            attribs.update({prop: getattr(self, prop) for prop in props})
+        return attribs
```

Comment:
This approach was suggested in a pydantic issue when using pydantic v1 (https://github.com/pydantic/pydantic/issues/935). Since we are in the process of moving to v2 but are not on v2 yet I've included it. When we move to v2 this can be replaced with the use of `computed_fields` https://docs.pydantic.dev/latest/usage/computed_fields/